qa/L0_trace/opentelemetry_unittest.py

```diff
@@ -153,121 +191,560 @@ def _check_events(self, span_name, events):
 
-    def _check_parent(self, child_span, parent_span):
-        # Check that child and parent span have the same trace_id
-        # and child's `parent_span_id` is the same as parent's `span_id`
-        self.assertEqual(child_span["trace_id"], parent_span["trace_id"])
-        self.assertNotEqual(
-            child_span["parent_span_id"],
-            NO_PARENT_SPAN,
-            "child span does not have parent span id specified",
+    def _test_resource_attributes(self, attributes):
+        """
+        Helper function that verifies passed span attributes.
+        Currently only test 2 attributes, specified upon tritonserver start:
+
+        --trace-config=opentelemetry,resource=test.key=test.value
+        and
+        --trace-config=opentelemetry,resource=service.name=test_triton
+
+        Args:
+            attributes (List[dict]): list of attributes, collected for a span.
+        """
+        expected_service_name = dict(
+            {"key": "service.name", "value": {"stringValue": "test_triton"}}
         )
+        expected_test_key_value = dict(
+            {"key": "test.key", "value": {"stringValue": "test.value"}}
+        )
+        self.assertIn(
+            expected_service_name,
+            attributes,
+            "Expected entry: {}, was not found in the set of collected attributes: {}".format(
+                expected_service_name, attributes
+            ),
+        )
+        self.assertIn(
+            expected_test_key_value,
+            attributes,
+            "Expected entry: {}, was not found in the set of collected attributes: {}".format(
+                expected_test_key_value, attributes
+            ),
+        )
+
+    def _verify_contents(self, spans, expected_counts):
+        """
+        Helper function that:
+         * iterates over `spans` and for every span it verifies that proper events are collected
+         * verifies that `spans` has expected number of total spans collected
+         * verifies that `spans` contains expected number different spans,
+           specified in `expected_counts` in the form:
+                    span_name : #expected_number_of_entries
+
+        Args:
+            spans (List[dict]): list of json objects, extracted from the trace and
+                   containing span info. For this test `name`
+                   and `events` are required.
+            expected_counts (dict): dictionary, containing expected spans in the form:
+                    span_name : #expected_number_of_entries
+        """
+        rex_name_field = re.compile("(?<='name': ')[a-zA-Z0-9_\- ]+(?=')")
+
+        span_names = []
+        for span in spans:
+            # Check that collected spans have proper events recorded
+            span_name = span[0]["name"]
+            span_names.append(span_name)
+            span_events = span[0]["events"]
+            event_names_only = rex_name_field.findall(str(span_events))
```

Comment:
Can you share sample event format that's getting parsed? Just curious what's being parsed by the regex.

Reply:
![image](https://github.com/triton-inference-server/server/assets/124622579/5818e79b-4429-4c49-b57f-b467cc1606d6)

Reply:
In this case it is easier to pass `span[0]["events"]` as a string and do regex.

`span[0]["events"]` is actually a list of dictionaries (on image it is seen better), that is why for this test I didn't want to parse `span[0]["events"]` list to collect all names.

Reply:
Isn't it same? 
![image](https://github.com/triton-inference-server/server/assets/21284872/4feb0b82-a923-4c08-8d2f-f1993d77611b)

Reply:
nice, bad on my part. Will remove `regex`